python/guide/module/gluon/init.md

```diff
@@ -161,3 +170,127 @@ net.add(nn.Dense(10, in_units=256))
+need to implement the `_init_weight` function and modify the incoming `NDArray`
+according to the initial result. In the example below, we  pick a decidedly
+bizarre and nontrivial distribution, just to prove the point. We draw the
+coefficients from the following distribution:
+
+$$
+\begin{aligned}
+    w \sim \begin{cases}
+        U[5, 10] & \text{ with probability } \frac{1}{4} \\
+            0    & \text{ with probability } \frac{1}{2} \\
+        U[-10, -5] & \text{ with probability } \frac{1}{4}
+    \end{cases}
+\end{aligned}
+$$
+
+```{.python .input  n=12}
+class MyInit(init.Initializer):
+    def _init_weight(self, name, data):
+        print('Init', name, data.shape)
+        data[:] = nd.random.uniform(low=-10, high=10, shape=data.shape)
+        data *= data.abs() >= 5
+
+net.initialize(MyInit(), force_reinit=True)
+net[0].weight.data()[0]
+```
+
+If this functionality is insufficient, we can even set parameters directly.
+Since `data()` returns an `NDArray` we can access it just like any other matrix.
+A note for advanced users - if you want to adjust parameters within an
+`autograd` scope you need to use `set_data` to avoid confusing the automatic
+differentiation mechanics.
+
+```{.python .input  n=13}
+net[0].weight.data()[:] += 1
+net[0].weight.data()[0,0] = 42
+net[0].weight.data()[0]
+```
+
+## Tied Parameters
+
+In some cases, we want to share model parameters across multiple layers. For
+instance when we want to find good word embeddings we may decide to use the
+same parameters both for encoding and decoding of words. We discussed one such
+case when we introduced `Blocks`. Let's see how to do
+this a bit more elegantly. In the following we allocate a dense layer and then
+use its parameters specifically to set those of another layer.
+
+```{.python .input  n=14}
+net = nn.Sequential()
+# we need to give the shared layer a name such that we can reference its parameters
+shared = nn.Dense(8, activation='relu')
+net.add(nn.Dense(8, activation='relu'),
+        shared,
+        nn.Dense(8, activation='relu', params=shared.params),
+        nn.Dense(10))
+net.initialize()
+
+x = nd.random.uniform(shape=(2, 20))
+net(x)
+
+# Check whether the parameters are the same
+print(net[1].weight.data()[0] == net[2].weight.data()[0])
+net[1].weight.data()[0,0] = 100
+# And make sure that they're actually the same object rather than just having the same value.
+print(net[1].weight.data()[0] == net[2].weight.data()[0])
+```
+
+The above exampe shows that the parameters of the second and third layer are
+tied. They are identical rather than just being equal. That is, by changing one
+of the parameters the other one changes, too. What happens to the gradients is
+quite ingenious. Since the model parameters contain gradients, the gradients of
+the second hidden layer and the third hidden layer are accumulated in the
+`shared.params.grad( )` during backpropagation.
```

Comment:
`shared.params.grad( )`: I don't think this is a method

Reply:
removed 'the'  - shared is a variable above.

Reply:
Was asking about `grad`. It's usually a property not a method.

Reply:
Done